libs/orders/src/lib/order-hooks/use-order-cancel.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useVegaWallet, useVegaTransaction } from '@vegaprotocol/wallet';
+import { useApolloClient } from '@apollo/client';
+import type {
+  OrderEvent,
+  OrderEventVariables,
+  OrderEvent_busEvents_event_Order,
+} from './__generated__/OrderEvent';
+import { ORDER_EVENT_SUB } from './order-event-query';
+import * as Sentry from '@sentry/react';
+import { OrderStatus } from '@vegaprotocol/types';
+import { determineId } from '@vegaprotocol/react-helpers';
+
+export const useOrderCancel = () => {
+  const { keypair } = useVegaWallet();
+  const { send, transaction, reset: resetTransaction } = useVegaTransaction();
+  const [updatedOrder, setUpdatedOrder] =
+    useState<OrderEvent_busEvents_event_Order | null>(null);
+  const [id, setId] = useState('');
+  const client = useApolloClient();
+
+  const reset = useCallback(() => {
+    resetTransaction();
+    setUpdatedOrder(null);
+    setId('');
+  }, [resetTransaction]);
+
+  useEffect(() => {
+    const clientSub = client.subscribe<OrderEvent, OrderEventVariables>({
+      query: ORDER_EVENT_SUB,
+      variables: { partyId: keypair?.pub || '' },
+    });
+
+    // Start a subscription looking for the newly created order
+    const sub = clientSub.subscribe(({ data }) => {
+      if (!data?.busEvents?.length) {
+        return;
+      }
+
+      // No types available for the subscription result
+      const matchingOrderEvent = data.busEvents.find((e) => {
+        if (e.event.__typename !== 'Order') {
+          return false;
+        }
+
+        return e.event.id === id;
+      });
+
+      if (
+        matchingOrderEvent &&
+        matchingOrderEvent.event.__typename === 'Order'
+      ) {
+        setUpdatedOrder(matchingOrderEvent.event);
+        resetTransaction();
```

Comment:
question (non-blocking): Could be done in a separate PR. But I think part of the idea of switching to use client.subscribe was so that we could store the subscription instance in a ref and start it and stop when a user executes a transaction, rather than using useEffect on mount. We would still need an effect to stop a running subscription on unmount. But we should only start it when we are actively awaiting an order.